[SPARK-36415][SQL][DOCS] Add docs for try_cast/try_add/try_divide (#33638)
apache/spark: docs/sql-ref-ansi-compliance.md

  - `map_col[key]`: This operator throws `NoSuchElementException` if key does not exist in map.
  - `GROUP BY`: aliases in a select list can not be used in GROUP BY clauses. Each column referenced in a GROUP BY clause shall unambiguously reference a column of the table resulting from the FROM clause.

> ### Special functions for using the ANSI dialect

Reviewer: Maybe ...

```suggestion
### Useful Functions for ANSI Mode
```
since the functions are not dependent on ANSI mode conf

> After turning ANSI mode on, if you expect some of your SQL operations to not throw exceptions on errors, as Spark's default behavior, you can use the following functions.

Reviewer: How about this?

```suggestion
When ANSI mode is on, it throws exceptions for invalid operations. You can use the following functions for SQL operations to suppress such exceptions as if ANSI mode is disabled.
```

Author: It's not "as if ANSI mode is disabled":
1. The acceptable input of TRY_CAST is the same as "CAST" in ANSI mode
2. When ANSI mode is disabled, `1+Int.MaxValue` is -2147483648, while `TRY_ADD(1, Int.MaxValue)` is null

  - `try_cast`: identical to `CAST`, except that it returns `NULL` result instead of throwing an exception on runtime error.
  - `try_add`: identical to the add operator `+`, except that it returns `NULL` result instead of throwing an exception on integral value overflow.
  - `try_divide`: identical to the division operator `/`, except that it returns `NULL` result instead of throwing an exception on dividing 0.

> Note that the behavior of these expressions doesn't depend on configuration `spark.sql.ansi.enabled`.

Reviewer: I think we can maybe remove this ...

Author: ok
